File: src/main/java/com/atsuishio/superbwarfare/item/gun/handgun/M1911Item.java
package com.atsuishio.superbwarfare.item.gun.handgun;

import com.atsuishio.superbwarfare.client.GunRendererBuilder;
import com.atsuishio.superbwarfare.client.model.item.M1911ItemModel;
import com.atsuishio.superbwarfare.item.gun.GunGeoItem;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.Rarity;
import software.bernie.geckolib.renderer.GeoItemRenderer;

import java.util.function.Supplier;

public class M1911Item extends GunGeoItem {

    public M1911Item() {
        super(new Properties().rarity(Rarity.COMMON));
    }

    @Override
    public Supplier<? extends GeoItemRenderer<? extends Item>> getRenderer() {
        return GunRendererBuilder.simple(M1911ItemModel::new);
    }

}

File: src/main/kotlin/com/atsuishio/superbwarfare/item/gun/handgun/M1911Item.kt
package com.atsuishio.superbwarfare.item.gun.handgun

import com.atsuishio.superbwarfare.data.gun.GunData
import com.atsuishio.superbwarfare.init.RegistryName
import com.atsuishio.superbwarfare.item.gun.GeoGunItemV2

@RegistryName("m_1911")
object M1911Item : GeoGunItemV2(Properties()) {
    override fun hasCustomMagazine(data: GunData): Boolean = true
    override fun hasCustomBarrel(data: GunData): Boolean = true
    override fun canEditAttachments(data: GunData): Boolean = true
}
